modules/python/src2/gen2.py

```diff
@@ -174,6 +174,30 @@
 gen_template_rw_prop_init = Template("""
     {(char*)"${member}", (getter)pyopencv_${name}_get_${member}, (setter)pyopencv_${name}_set_${member}, (char*)"${member}", NULL},""")
 
+gen_template_overloaded_function_call = Template("""
+    {
+${variant}
+
+        if (PyErr_Occurred() != NULL)
+        {
+            PySafeObject exception_type;
+            PySafeObject exception_value;
+            PySafeObject exception_traceback;
+            PyErr_Fetch(exception_type, exception_value, exception_traceback);
+            PyErr_NormalizeException(exception_type, exception_value, exception_traceback);
+
+            PySafeObject exception_message(PyObject_Str(exception_value));
+            std::string message;
+            getUnicodeString(exception_message, message);
+            #ifdef CV_CXX11
+                conversion_errors.push_back(std::move(message));
+            #else
+                conversion_errors.push_back(message);
+            #endif
+        }
```

Comment:
Lets keep generated code minimal. Move this logic to dedicated function in cv2.cpp.

See this patch: #15951